File: library/src/main/java/cn/wandersnail/http/ConvertedResponse.java
package cn.wandersnail.http;

import androidx.annotation.Nullable;

import okhttp3.Response;
import okhttp3.ResponseBody;
import retrofit2.Call;

/**
 * 转换过的响应数据
 * 
 * date: 2019/8/23 20:51
 * author: zengfansheng
 */
public class ConvertedResponse<T> {
    private final Call<ResponseBody> call;
    /**
     * 转换过的响应数据
     */
    @Nullable
    public T convertedResponse;
    /**
     * 转换异常
     */
    @Nullable
    public Throwable convertError;
    /**
     * 原始响应
     */
    @Nullable
    public Response raw;
    /**
     * 是否请求超时了
     */
    public boolean isCallTimeout;

    public ConvertedResponse(Call<ResponseBody> call) {
        this.call = call;
    }

    /**
     * 取消
     */
    public void cancel() {
        if (call.isExecuted() && !call.isCanceled()) {
            call.cancel();
        }
    }

    /**
     * 是否被取消了
     */
    public boolean isCanceled() {
        return call.isCanceled();
    }
}

File: library/src/main/java/com/snail/network/ConvertedResponse.kt
package com.snail.network

import okhttp3.Response
import okhttp3.ResponseBody
import retrofit2.Call

/**
 * 转换过的响应数据
 *
 * date: 2019/5/16 20:08
 * author: zengfansheng
 */
class ConvertedResponse<T> internal constructor(private val call: Call<ResponseBody>) {
    var convertedBody: T? = null
    /**
     * 转换异常
     */
    var convertRrror: Throwable? = null
    /**
     * 原始响应
     */
    var raw: Response? = null
    /**
     * 是否请求超时了
     */
    var callTimeout = false
    /**
     * 是否被取消了
     */
    val isCanceled
        get() = call.isCanceled

    /**
     * 取消
     */
    fun cancel() {
        if (call.isExecuted && !call.isCanceled) {
            call.cancel()
        }
    }
}